### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by aflegel**

### CODE REVIEW

File: DataFramework/Context/ProbabilityContext.Accessors.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		/// Returns the id for the pool with the matching dice, this is the hard way
+		/// </summary>
+		/// <param name="searchForPool"></param>
+		/// <returns></returns>
+		public static int? GetPoolId(ProbabilityContext context, List<PoolDie> searchForPool) =>
+			searchForPool.Select(die => context.PoolDice.Where(w => w.DieId == die.DieId && w.Quantity == die.Quantity && w.Pool.PoolDice.Count == searchForPool.Count())
+			.Select(s => s.PoolId)).Aggregate((result, next) => result.Intersect(next)).FirstOrDefault();
+
+		/// <summary>
+		/// Trys to find the pool id from a given name
```
Comment (ChrisDusyk) [inline DataFramework/Context/ProbabilityContext.Accessors.cs]: `Trys` spelling mistake

File: DataFramework/Context/ProbabilityContext.Accessors.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		/// </summary>
+		/// <param name="context"></param>
+		/// <param name="poolName"></param>
+		/// <returns></returns>
+		public static int? GetPoolIdByName(this ProbabilityContext context, string poolName) => context.Pools.Where(w => w.Name == poolName).Select(s => s.PoolId).FirstOrDefault();
+
+		/// <summary>
+		/// Trys to get the pool ids split by positive and negative dice
```
Comment (ChrisDusyk) [inline DataFramework/Context/ProbabilityContext.Accessors.cs]: `Trys` again 🤔

File: DataFramework/Models/PoolResult.cs [hunk 3]
```diff
@@ -44,5 +26,7 @@ public PoolResult(List<PoolResultSymbol> PoolResultSymbols)
 		/// <param name="key"></param>
 		/// <returns></returns>
 		public int CountMatchingKeys(Symbol key) => PoolResultSymbols.Where(a => a.Symbol == key).Sum(s => s.Quantity);
+
+		public override int GetHashCode() => PoolResultSymbols.Select(s => s.GetHashCode()).Aggregate(1, (a, b) => unchecked(a * b));
```
Comment (ChrisDusyk) [inline DataFramework/Models/PoolResult.cs:30]: This is clever

File: Visualizer.Tests/APITests.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+		[Fact]
+		public void ResultsPositive()
+		{
+			var controller = new RollController(context);
+
+			var result = controller.Get(PositiveModel.Dice.ToList());
+			Assert.True(15 == result.PositiveRolls.Results.Count(), $"Statistics count incorrect.  Count was {result.PositiveRolls.Results.Count()}");
+			Assert.True(15 == result.NegativeRolls.Results.Count(), $"Statistics count incorrect.  Count was {result.PositiveRolls.Results.Count()}");
+
+			controller.Dispose();
```
Comment (ChrisDusyk) [inline Visualizer.Tests/APITests.cs]: Is there a specific reason to `Dispose` of these controllers explicitly, vs letting GC handle them?

File: Visualizer.Tests/SeedingTests.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+			Assert.True(1 == pools.Item1.Count(), $"Incorrect number of results {pools.Item1.Count()}");
+			Assert.True(15 == pool.PoolResults.Count, $"The number of results did not equal 15. Result was {pool.PoolResults.Count}");
+			Assert.True(64 == pool.RollEstimation(), $"The total outcomes did not equal 64. Result was {pool.PoolResults.Count}");
+		}
+
+		[Fact]
+		public void DieComparisonGeneratorAdvanced()
+		{
+			var pools = DiceSeed.SeedDice().ProcessPools((Enumerable.Range(1, 2), Enumerable.Range(1, 1), Enumerable.Range(0, 1)), (Enumerable.Range(1, 2), Enumerable.Range(1, 1), Enumerable.Range(0, 1))).CrossProduct();
+			//var pool = pools.First();
```
Comment (ChrisDusyk) [inline Visualizer.Tests/SeedingTests.cs]: Cleanup

File: Visualizer/Models/DieViewModel.cs [hunk 6]
```diff
@@ -1,9 +1,19 @@
...
+namespace Visualizer.Models
 {
 	public class DieViewModel
 	{
 		public string DieType { get; set; }
 
 		public int Quantity { get; set; }
 	}
+
+	public static class DieViewModelExtensions
```
Comment (ChrisDusyk) [inline Visualizer/Models/DieViewModel.cs]: Might want to split this into its own file?

**Review by ChrisDusyk — APPROVED** [review]
This was a lot of review! So I think you've done a good job refactoring, though it's kind of tough to review a refactor that well.

Something you might want to look into for your tests is [`FluentAssertions`](https://fluentassertions.com/). It'll make them nicer to read.

File: Visualizer.Tests/APITests.cs (hunk 4 above)
Comment (aflegel) [reply Visualizer.Tests/APITests.cs]: It complained at me actually.  

I think this is normally hooked up as part of the services but running it in a test it doesn't get the normal treatment.

**Review by ChrisDusyk — APPROVED** [review]
I think those were some good changes, especially for the tests

**Pull request merged**